### PR TITLE
feat(storage): add object retention samples

### LINKED ISSUE #3803: storage: add object retention samples
Please add the following samples for object retention, to go on the [docs page](https://cloud.google.com/storage/docs/using-object-lock):

`storage_create_bucket_with_object_retention`
`storage_set_object_retention_policy`

Also, please update `storage_get_bucket_metadata` and `storage_get_metadata ` to return object retention related metadata.

Please use the [Java samples](https://github.com/googleapis/java-storage/pull/2417) as a canonical reference.

### PR DESCRIPTION
**Pull request by BrennaEpp**
## Description

Fixes #3803 

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: storage/objects/get_metadata.go [hunk 1]
```diff
@@ -64,6 +64,10 @@ func getMetadata(w io.Writer, bucket, object string) (*storage.ObjectAttrs, erro
 	fmt.Fprintf(w, "Temporary hold enabled? %t\n", attrs.TemporaryHold)
 	fmt.Fprintf(w, "Retention expiration time %v\n", attrs.RetentionExpirationTime)
 	fmt.Fprintf(w, "Custom time %v\n", attrs.CustomTime)
+	if attrs.Retention != nil {
```
Comment (tritone) [inline storage/objects/get_metadata.go:67]: IMO just print this and remove the conditional.  Use `%+v` to include the field names in the struct.

File: storage/objects/objects_test.go [hunk 2]
```diff
@@ -631,3 +631,41 @@ func TestObjectBucketLock(t *testing.T) {
...
+	if err != nil {
+		t.Errorf("setObjectRetention: %v", err)
+	}
+	attrs, err := bucket.Object(objectName).Attrs(ctx)
+	if err != nil {
+		t.Errorf("object.Attrs: %v", err)
+	}
+
+	if attrs.Retention == nil {
+		t.Errorf("mismatching retention config, wanted %+v, got nil", attrs.Retention)
```
Comment (tritone) [inline storage/objects/objects_test.go:669]: Nit: got before want. Also we should probably check that it's the expected number of days as well (well, under that number given the test execution time).

**Review by tritone — APPROVED** [review]
Two minor nits, otherwise LGTM

File: storage/objects/get_metadata.go (hunk 1 above)
Comment (tritone) [reply storage/objects/get_metadata.go:67]: Eh, I don't think it matters that much. Honestly we should probably just remove all the specific fields from the sample and use `%+v` on the Attrs object (C++ does this...).

**Review by tritone — APPROVED** [review]

**Comment by muncus** [comment]
Apologies for the struggle with automerge. i've examined the failing tests, and they do not appear related to this change.

**Pull request merged**